commcare-cloud/commcare_cloud.py

```diff
@@ -0,0 +1,125 @@
+# coding=utf-8
+from __future__ import print_function
+import getpass
+import os
+from six.moves import input
+from argparse import ArgumentParser
+import subprocess
+from clint.textui import puts, colored
+
+
+def ask(message):
+    return 'y' == input('{} [y/N]'.format(message))
+
+
+class AnsiblePlaybook(object):
+    @staticmethod
+    def make_parser(parser):
+        parser.add_argument('--skip-check', action='store_true', default=False)
+        parser.add_argument('--branch', default='master')
+        parser.add_argument('playbook')
+
+    @staticmethod
+    def run(args, unknown_args):
+        check_branch(args)
+
+        def ansible_playbook(environment, playbook, vault_password, *cmd_args):
+            cmd = (
+                'ANSIBLE_CONFIG={}'.format(os.path.expanduser('~/.commcare-cloud/ansible/ansible.cfg')),
+                'ansible-playbook',
+                os.path.expanduser('~/.commcare-cloud/ansible/{playbook}'.format(playbook=playbook)),
+                '-u', 'ansible',
+                '-i', os.path.expanduser('~/.commcare-cloud/inventory/{env}'.format(env=environment)),
+                '-e', '"@{}"'.format(os.path.expanduser('~/.commcare-cloud/vars/{env}/{env}_vault.yml'.format(env=environment))),
+                '-e', '"@{}'.format(os.path.expanduser('~/.commcare-cloud/vars/{env}/{env}_public.yml"'.format(env=environment))),
+                '--vault-password-file=/bin/cat',
+                '--diff',
+            ) + cmd_args
+            print(' '.join(cmd))
+            p = subprocess.Popen(' '.join(cmd), stdin=subprocess.PIPE, shell=True)
```

Comment:
It would be safer to use `' '.join(shlex_quote(arg) for arg in cmd)` (and `from six.moves import shlex_quote`) instead of `' '.join(cmd)`. As is will break if any of the arguments contain spaces.

Reply:
Thanks good call: https://github.com/dimagi/commcarehq-ansible/pull/1038